x/metadata/keeper/query_server.go

```diff
@@ -649,6 +650,38 @@ func (k Keeper) ScopeSpecification(c context.Context, req *types.ScopeSpecificat
 		retval.ScopeSpecification = types.WrapScopeSpecNotFound(specAddr)
 	}
 
+	if found && req.IncludeContractSpecs {
+		err := k.IterateContractSpecs(ctx, func(cs types.ContractSpecification) (stop bool) {
+			for _, id := range spec.ContractSpecIds {
+				if bytes.Equal(id, cs.SpecificationId) {
+					retval.ContractSpecs = append(retval.ContractSpecs, types.WrapContractSpec(&cs))
+				}
+			}
+			return false
+		})
+		if err != nil {
+			return &retval, fmt.Errorf("error iterating scope spec [%s] contract specs: %w", spec.SpecificationId, err)
+		}
+	}
+
+	if found && req.IncludeRecordSpecs {
+		var err error
+		for _, id := range spec.ContractSpecIds {
+			err = k.IterateRecordSpecsForContractSpec(ctx, id, func(recordSpecID types.MetadataAddress) (stop bool) {
+				err = k.IterateRecordSpecs(ctx, func(recordSpec types.RecordSpecification) (stop bool) {
```

Comment:
Whoa. Way too much work here. You've got the record id, just get it directly: `k.GetRecordSpecification(ctx, recordSpecID)`.

Reply:
changed to use `k.GetRecordSpecification` for better performance.